sphinx/texinputs/sphinx.sty

```diff
@@ -282,6 +255,72 @@
   % LaTeX environments always revert local changes on exit, here e.g. \parskip
 }
 
+% For shadowing boxes, with allowed page breaks
+\def\Sphinx@shadowfboxsep {5\p@}% 5pt used to be hard-coded in latex.py
+\def\Sphinx@shadowsize    {4\p@}% 4pt was the default for \shadowbox from fancybox.sty
+
+\long\def\Sphinx@ShadowFBox#1{%
+  \leavevmode\begingroup
+  \setbox\@tempboxa
+   \hbox{\vrule\@width\fboxrule
+         \vbox{\hrule\@height\fboxrule
+               \kern\Sphinx@shadowfboxsep
+               \hbox{\kern\Sphinx@shadowfboxsep #1\kern\Sphinx@shadowfboxsep}%
+               \kern\Sphinx@shadowfboxsep
+               \hrule\@height\fboxrule}%
+         \vrule\@width\fboxrule }%
+  % we imitate what \shadowbox from fancybox.sty would do
+  \dimen@\dimexpr.5\fboxrule+\Sphinx@shadowsize\relax
+  \hbox{\vbox{\offinterlineskip
+        \hbox{\copy\@tempboxa
+              \kern-.5\fboxrule
+              \lower\Sphinx@shadowsize
+              \hbox{\vrule\@height\ht\@tempboxa \@width\dimen@}}%
+        \kern-\dimen@
+        \moveright\Sphinx@shadowsize\vbox{\hrule\@width\wd\@tempboxa \@height\dimen@}%
+        }\kern-\Sphinx@shadowsize % shadow on right does not add to width
+       }%
+  \endgroup
+}
+
+\newenvironment{SphinxShadowBox}
+  {\def\FrameCommand {\Sphinx@ShadowFBox }%
+   % earlier use from latex.py  of "center" and a \vbox was buggy and caused
+   % vertical space which we try to imitate as much as possible via "trivlist"
+   \OuterFrameSep \z@skip
+   \trivlist\item\relax\null % makes blank line on top to imitate <= 1.4.1
```

Comment:
Sphinx's latex.py used a center environment with  a box wider than text width. This causes extra vertical space compared to something which would fit in textwidth. It is difficult to reconstitue that spacing, hence we use \trivlist like \center does to emulate it. But there is a subtlety then than MakeFramed does not quite vertical mode. At the end when \endtrivlist is found, it looks as if vertical mode was not left, which LaTeX interprets as no contents after \item. In such case it inserts extra blank line. But in case of topic box for tables of contents, there was use of \itemize, and this toggles booleans, so that LaTeX in that case does not think there was nothing after \item, and as a result does not insert the extra blank line at time of \endtrivlist. This never occurred earlier because \shadowbox quit vertical mode, which framed does not do.